modnet/fit_genetic.py

```diff
@@ -93,29 +159,37 @@ def crossover(
             father: List containing the gentic information of the second parent.
         """
 
-        child = [mother[0], father[1], mother[2], father[3], mother[4], father[5], mother[6], father[7], mother[8], father[9]]   
+        genes_from_mother = random.sample(range(10), k=5)
+        child = [mother[i] if i in genes_from_mother else father[i] for i in range(10)]   
         return child
 
 
     def mutation(
         self,
         child: List,
-        prob_mut: float = 0.8
         )->None:
 
         """Performs mutation in the genetic information in order to maintain diversity in the population. 
         
         Paramters:
             child: List containing the genetic information of the 'child'.
-            prob_mut: Probability the mutation occurs.
         """
 
         for c in range(0, len(child)):
-            if np.random.rand() > prob_mut:
-                if child[c][0] < int(0.9*len(self.X_train.get_optimal_descriptors())):
-                    child[c][0] = int(child[c][0] + 10*randint(1, 6))
-                else:
-                    child[c][0] = int(child[c][0] - 10*randint(1, 6))
+            if child[c][0] < int(0.5*len(self.data.get_optimal_descriptors())):
+                child[c][0] = int(child[c][0] + randint(1, int(0.1*len(self.data.get_optimal_descriptors()))))
+                child[c][1] = child[c][1] + 32*randint(-2,2)
+                child[c][2] = random.choice(self.fraction)
+                child[c][3] = random.choice(self.fraction)
+                child[c][4] = random.choice(self.fraction)
+                child[c][8] = random.choice(self.lr)
+            else:
+                child[c][0] = int(child[c][0] - randint(1, int(0.1*len(self.data.get_optimal_descriptors()))))
+                child[c][1] = child[c][1] + 32*randint(-2,2)
+                child[c][2] = random.choice(self.fraction)
+                child[c][3] = random.choice(self.fraction)
+                child[c][4] = random.choice(self.fraction)
+                child[c][8] = random.choice(self.lr)
```

Comment:
All childs will converge between 0.4 and 0.6 times number of features, which seems quite limited to me.